### PR TITLE
[polaris-migrator] Add a migration to change interpolation in media queries by env

### PR DESCRIPTION
**Pull request by jbalsas**
### WHY are these changes introduced?

In the context of `shopify/web`'s code yellow, we are looking to drop sass.

### WHAT is this pull request doing?

Creates a new migration `scss-replace-media-query-interpolation` that replaces breakpoint interpolation in media queries by [env()](https://developer.mozilla.org/en-US/docs/Web/CSS/env) which can be used there natively.

In web, we will be providing the breakpoints through postcss for now using [poscss-env-function](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-env-function)

In essence, the migration turns this `scss`

```scss
@use 'global-styles/media-queries';

.EmptyPage {
  max-width: 250px;

  @media #{media-queries.$p-breakpoints-md-down} {
    max-width: 300px;
  }
}
```

into this `css`

```css
.EmptyPage {
  max-width: 250px;

  @media (env(--p-breakpoints-md-down)) {
    max-width: 300px;
  }
}
```

I'd also love to get 👀 from the polaris team about this approach. Are we missing something here? If we think it works, we could expose the breakpoints through JS and even share them across css and js. We could also update the docs on [breakpoints](https://polaris.shopify.com/tokens/breakpoints)

### How to 🎩

I think tests should be enough, but you can look at https://github.com/Shopify/web/pull/105405/commits/184829d27da0b80ccf320ab7d76722027f73ffb9 as a reference of the output of running the codemod over a fraction of the `shopify/web` codebase

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-migrator/src/migrations/scss-replace-media-query-interpolation/transform.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import type {API, FileInfo, Options} from 'jscodeshift';
+import type {Plugin} from 'postcss';
+import postcss from 'postcss';
+import stylelint from 'stylelint';
+
+const plugin = (): Plugin => {
+  return {
+    postcssPlugin: 'scss-replace-media-query-interpolation',
+    AtRule(atRule) {
+      if (atRule.name === 'media' || atRule.name === 'container') {
```
Comment (jbalsas) [inline polaris-migrator/src/migrations/scss-replace-media-query-interpolation/transform.ts:10]: Didn't dive too deep into the API, so wondering if there's a 'proper' way to do this.

**Review by kyledurand — APPROVED** [review]
I'm down with this approach. The env() api is pretty clunky, but so was sass, and it solves a hard problem.

If anyone is interested: I put up a [draft PR](https://github.com/Shopify/polaris/pull/10786) for how this could work on our end, and create and export for web to get a better sense of how we'd work together.

**Review by mateus — APPROVED** [review]

**Comment by sam-b-rose** [comment]
The codemod looks great! Do you have any info on the decision to use `env` versus something like [`postcss-design-tokens`](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-design-tokens)? Seems like that is what the README [was suggesting](https://github.com/csstools/postcss-plugins/blob/main/plugins/postcss-env-function/README.md?plain=1#L13) over `postcss-env-function`.

**Comment by jbalsas** [comment]
Hey @sam-b-rose, good observation!

I realized the same this morning as our prod build would fail with these changes 😭 

> Closing for now. This is failing prod build because importFrom is supposed to be deprecated since it's considered a side effect. See https://github.com/csstools/postcss-plugins/discussions/192 for more info 😢
> 
> I'll analyze the more commonly suggested alternative which seems to be [postcss-design-tokens](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-design-tokens).

I think this feels like more forward-looking and "standardized" solution than what's proposed in the [design-tokens](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-design-tokens) plugin, but we might need to go with that after all. Going to close this one for now until we settle on something final! 🙏

**Pull request closed without merging**